client/highlevel-rest/build.gradle

```diff
@@ -36,26 +31,30 @@ dependencies {
   testCompile "org.hamcrest:hamcrest-all:${versions.hamcrest}"
 }
 
-dependencyLicenses {
-  // Don't check the client's license. We know it.
-  dependencies = project.configurations.runtime.fileCollection {
-    it.group.startsWith('org.elasticsearch') == false
+task checkNoOtherDependencies {
+  description 'This task fails if any other dependency than core and rest client is added'
+  group 'Verification'
+  doLast {
+    Collection<Dependency> others = configurations.compile.allDependencies - configurations.compile.allDependencies.findAll {
+      // Should only contain core and rest client!
+      (it.group == project.group && it.name == 'rest') || (it.group == 'org.elasticsearch' && it.name == 'elasticsearch')
+    }
+    if (others.isEmpty() == false) {
+      throw new GradleException(project.path + " must depend only on core and rest client, but found: " + others)
+    }
   }
 }
 
-//JarHell is part of es core, which we don't want to pull in
-jarHell.enabled=false
-
-task integTest(type: RestIntegTestTask,
-                  group: JavaBasePlugin.VERIFICATION_GROUP,
-                  description: 'Multi-node tests',
-                 dependsOn: test.dependsOn) {
-     configure(BuildPlugin.commonTestConfig(project))
-     classpath = project.test.classpath
-     testClassesDir = project.test.testClassesDir
-     include '**/*IT.class'
```

Comment:
It seems you removed this, but I would guess you meant to call the method you extracted out in RestTestPlugin?  Instead of importing a class to call a method which creates tasks, create alternate gradle plugins (furthing the composition that is already there). For example, create a IntegTestPlugin, which RestTestPlugin will either extend or wrap (there are other places where we apply `elasticsearch.rest-test` just to have integ tests so having `elasticsearch.integ-test` would be good).